### PR TITLE
Fix sending collectible by building transaction before signing

### LINKED ISSUE #21702: "cannot resolve route id" error is shown after an attempt to send collectible
### Steps:
1. Go to collectible sending flow
2. Send collectible

### Actual result:
Collectible is not sent. Toast with "cannot resolve route id" error is shown

### Expected result:
Collectible is sent

### Logs:
[Status-debug-logs (6).zip](https://github.com/user-attachments/files/17947948/Status-debug-logs.6.zip)

https://github.com/user-attachments/assets/962ca67d-93f6-46df-9679-ac884c125a92

### Additional info:
For IOS send button is shown just as disabled
image

### PR DESCRIPTION
**Pull request by vkjr**
fixes #21702

### Summary

After migrating to new endpoint for sending transaction the `build transaction` step wasn't added to collectibles send flow which resulted in inability to send transaction. This PR fixes the problem.

### Review notes
The fix looks "dirty" and probably more thought can be put in the way we organise events to have consistency between sending tokens and collectibles, but for pre-release "duct tape fix" I believe should be ok.

### Testing notes
Here is the transaction I've made with the help of fix.
https://sepolia.etherscan.io/tx/0x6b7b5fa210839e668f623af5117e86cf1b67d8fd1c80cc4ecf810bd8fe87048e

#### Platforms

- Android
- iOS

##### Functional

- wallet / transactions

### Steps to test
- Open Status
- Follow send collectible flow to reach confirmation screen
- Ensure that "slide to send" button is enabled, signing is possible and transaction can be send

status: ready

### CODE REVIEW

**Review by shivekkhurana — APPROVED** [review]
Looks good, thank you )

File: src/status_im/contexts/wallet/send/transaction_confirmation/view.cljs [hunk 1]
```diff
@@ -249,6 +249,16 @@
...
                                        :address   (utils/get-shortened-address
                                                    to-address)}]
+        ;; In token send flow we already have transaction built when
+        ;; we reach confirmation screen. But in send collectible flow
+        ;; routes request happens at the same time with navigation to
+        ;; confirmation screen. So we need to build the transaction as soon
+        ;; as route is available.
+        (rn/use-effect
+         (fn []
+           (when (and (= transaction-type :tx/collectible-erc-1155) first-route)
```
Comment (smohamedjavid) [inline src/status_im/contexts/wallet/send/transaction_confirmation/view.cljs]: Quick question: why not `:tx/collectible-erc-721`  is added?

Comment (vkjr) [reply src/status_im/contexts/wallet/send/transaction_confirmation/view.cljs]: @smohamedjavid, great catch, thank you!

**Comment by churik** [comment]
let us know when it is ready to be tested, thx!

**Comment by vkjr** [comment]
@churik, it is ready

File: src/status_im/contexts/wallet/send/transaction_confirmation/view.cljs (hunk 1 above)
Comment (smohamedjavid) [reply src/status_im/contexts/wallet/send/transaction_confirmation/view.cljs]: No worries 🙌 

Sorry for the last minute comment. For collectible tx type check there is a util method. Please use it 🙏 

https://github.com/status-im/status-mobile/blob/2be78041e2b5e0728a603155aaeeeacf427efff2/src/status_im/contexts/wallet/send/utils.cljs#L292-L294

**Review by clauxx — APPROVED** [review]

File: src/status_im/contexts/wallet/send/transaction_confirmation/view.cljs (hunk 1 above)
Comment (vkjr) [reply src/status_im/contexts/wallet/send/transaction_confirmation/view.cljs]: @smohamedjavid, ideal! thank you :)

**Review by smohamedjavid — APPROVED** [review]

**Comment by pavloburykh** [comment]
@vkjr thank you for the fix. PR is ready for merge.

**Pull request merged**